main/lsp/ExtractVariable.cc

```diff
@@ -93,6 +78,7 @@ class LocSearchWalk {
     // the current top of the stack as the "deepest" class/method
     vector<ast::ExpressionPtr *> enclosingClassStack;
     vector<ast::ExpressionPtr *> enclosingMethodStack;
+    bool inCsend = false;
```

Comment:
Usually keeping track of a boolean in these sorts of things falls over if there's ever some sort of nesting.

Does the change here handle cases like

```ruby
(a&.foo)&.bar
```

?

Reply:
Good catch, I updated it to use a counter instead, and added a few test cases for it.